### PR TITLE
Add example of docker image for theia-trace-ext

### LINKED ISSUE #908: Launching the trace viewer using docker
Is there a way to launch the trace viewer in browser using docker? 
Similar to how the Gitpod demo works, but without the Gitpod.

### PR DESCRIPTION
**Pull request by frallax**
Resolves #908

Signed-off-by: Francesco Robino <francesco.robino@ericsson.com>

### CODE REVIEW

File: examples/docker/README.md
Comment (marco-miller) [inline examples/docker/README.md:36]: `another`, rather.

File: examples/docker/README.md
Comment (marco-miller) [inline examples/docker/README.md:41]: If I run a trace-server locally as `http://localhost:8080/tsp/api`, non-containerized but say from an Eclipse debug config, then I'm unable to get line 28's IP. I rather get an empty output for that command. I tried with a few variations of the command right above (line 41's), without success. But WDYT?

**Review by marco-miller — CHANGES_REQUESTED** [review]
Awesome work, thank you @frallax! -Some comments, mostly minor except for my inability to get that trace-server connection it seems.

File: examples/docker/README.md
Comment (marco-miller) [reply examples/docker/README.md:41]: Nevermind, the viewer IP is `localhost`('s) in this case. And the container doesn't reveal that since trivial (or per `host` network definition) in its `inspect` output. Maybe clarify this in the README, around here or so?

**Comment by marco-miller** [comment]
> Awesome work, thank you @frallax! -Some comments, mostly minor except for my inability to get that trace-server connection it seems.

(The latter issue has been resolved since.)

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md:17]: I do not understand what you mean. When did you have to run the ```npm install``` command above? In order to run the application in a "dockerized" env I am expecting that you should have only docker as dependency... or am I misunderstanding?

File: examples/docker/example-package.json
Comment (frallax) [reply examples/docker/example-package.json:1]: There is a reason why this file is not named ```package.json```. If I rename it as you propose, when building the theia-trace-extension application from the source of this repo (I mean if you try to checkout this branc/pull-request, then you locally on your machine run ```cd theia-trace-extension && yarn```), it will create a lockfile that contains the dependency to  ``` "theia-traceviewer": "next" ```.

This probably happens because the package manager is looking recursively to all package.json in the project (including the ones in the example folder) and using them to build the dependency tree. I am not sure if this is expected or not, but I do not want that an "example" is going to pollute the dependency tree of the main project. Hence the renaming.

Please let me know if you have comments about this, but I think that fixing the issue (or feature?) of dependencies / ignoring or not the package.json files under the example folder should be kept in a separate change because this is not connected to the "dockerizing" work that this PR is targeting.

Comment (marco-miller) [reply examples/docker/example-package.json:1]: Ah, LGTM then too; thanks for clarifying. I'd add a note about this in the commit message though, to capture this important info in git history, up to you. Or in the README, at will. (Maybe the latter would be closer to us potential users of this.)

File: examples/docker/README.md
Comment (marco-miller) [reply examples/docker/README.md:17]: Not sure what to say or add here, as I did have to run the above `install` command the first time I tried this. If you didn't, or if nobody else did (beside me/my local environment), then let's ignore this for now, at will.

File: examples/docker/example-package.json
Comment (frallax) [reply examples/docker/example-package.json:1]: Done, please resolve if satisfied

File: examples/docker/example-package.json
Comment (frallax) [reply examples/docker/example-package.json:40]: I am not sure if those should be explained. I mean, for the other examples I am not sure it is explained somewhere why some params were used. I just thought I'd like to have a "production ready"-like image.

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md]: Done, please resolve if satisfied

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md]: Done, please resolve if satisfied

Comment (frallax) [reply examples/docker/README.md]: Done, please resolve if satisfied

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md]: Done, please resolve if satisfied

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md:1]: Done, please resolve if satisfied

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md:26]: Done, please resolve if satisfied

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md]: Done, please resolve if satisfied

File: examples/docker/Dockerfile
Comment (frallax) [reply examples/docker/Dockerfile]: Done, please resolve if satisfied

File: examples/docker/Dockerfile
Comment (frallax) [reply examples/docker/Dockerfile]: Done, please resolve if satisfied

File: examples/docker/README.md
Comment (frallax) [reply examples/docker/README.md]: Done, please resolve if satisfied

**Review by marco-miller — DISMISSED** [review]
LGTM now, thanks again. I'll get a second committer review before merging; WIP.

File: examples/docker/example-package.json
Comment (bhufmann) [inline examples/docker/example-package.json:18]: The version of the `theia-trace-extension` to 1.34.2 was updated in a recent commit. The version needs to be updated here as well. Otherwise the application can fail to start with the error reported [here](https://community.theia-ide.org/t/cannot-apply-injectable-decorator-multiple-times/2628). I got that error after fetching the PR and rebasing it. Could you please update the example-package.json for Theia 1.34.2 and rebase your code to the latest in master and update the PR? I tired it locally and it worked fine.

**Review by bhufmann — CHANGES_REQUESTED** [review]
Please update the Theia version in the example-packages.json

File: examples/docker/example-package.json
Comment (frallax) [reply examples/docker/example-package.json]: Yes, this is an issue due to the fact that the theia-traceviewer npm package has no versioning, hence this file has always to be maintained to the "latest and greatest". The main goal of this commit is just to show how we can create a docker image of the theia-trace-extension project.

If we want to avoid issues as the one you report above, we should:
- include the building and testing of the docker image (including starting it) in the CI machinery
- start versioning theia-trace-extension and the theia-traceviewer npm package
- just add in the readme a description of how to handle the issue you reported above

Initially I had a script to copy the package.json from the browser folder, and create the example-package.json, but I thought it was ugly (also if that would solve the issue of "keeping up to date").

Let me know if I can do something more to improve this PR, otherwise I think it is good enough to have an example and it will be up to us to keep it up to date when we uplift.

Comment (bhufmann) [reply examples/docker/example-package.json]: Thanks for the update. All good. We'll also update the Theia version in the example-packages.json when upgrading in the future.

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks a lot!

File: examples/docker/example-package.json
Comment (OljaH) [reply examples/docker/example-package.json]: Regarding this, do you have any advice on how I could make a stable app (set to a particular theia-trace-extension version) so that it doesn't crash every time theia-trace-extension is updated? Would putting _latest_ instead of a specific version do the job or is the only solution to write the script that @frallax mentioned?

Comment (marco-miller) [reply examples/docker/example-package.json]: The thing is, `theia-trace-extension` itself (published as `theia-traceviewer`) has no versioning, at all. Hence the necessary use of `next`, here. Now, the core theia libraries used internally across this whole `theia-trace-extension` repo (Docker parts included) are all pinned to `1.34.2` currently, which should be consistent enough. Goal being, that we keep maintaining all of these versions accordingly over time. We cannot use latest or next, because we need to pin a version of those libraries that is known to work as latest with this current repo codebase.

This should suffice, but anyone remains [welcome to contribute such maintenance acts](https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/master/CONTRIBUTING.md#uplifting-the-theia-dependencies-version), over time. While writing this, I just noticed that we [should mention docker now under here](https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/master/CONTRIBUTING.md#uplifting-the-theia-dependencies-version). So I'll push a PR to fix this soon.

Comment (marco-miller) [reply examples/docker/example-package.json]: <=> PR #954 that I just pushed for review (a simple one).

**Pull request merged**
